internal/monitor/exporter.go

```diff
@@ -106,3 +109,59 @@ func CloseOpenTelemetryCollectorExporter() {
 	}
 	ocExporter = nil
 }
+
+var prometheusExporter *prometheus.Exporter
+var prometheusServer *http.Server
+
+// EnablePrometheusCollectorExporter starts exporting monitoring metrics for
+// the Prometheus to scrape on the given port.
+func EnablePrometheusCollectorExporter(port string) error {
+	if port == "" {
+		return nil
+	}
+
+	var err error
+	if prometheusExporter, err = prometheus.NewExporter(
+		prometheus.Options{
+			OnError: func(err error) {
+				logger.Errorf("Fail to collect metric: %v", err)
+			},
+		},
+	); err != nil {
+		return fmt.Errorf("create Prometheus collector exporter: %w", err)
+	}
+
+	view.RegisterExporter(prometheusExporter)
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/metrics", prometheusExporter.ServeHTTP)
+	prometheusServer = &http.Server{
+		Addr:           ":" + port,
+		Handler:        mux,
+		ReadTimeout:    10 * time.Second,
+		WriteTimeout:   10 * time.Second,
+		MaxHeaderBytes: 1 << 20,
+	}
+
+	go func() {
```

Comment:
Though this should be lightweight but we should do run perf testing of GCSfuse with promotheus server running and also ensure that this doesn't take extra memory and CPU

Reply:
Hi @sethiay, thanks for the review! I updated the PR to move the mount flag to config file. Do you happen to have a guidance to do the GCSFuse perf test and memory/CPU resource tests?

Reply:
As we discussed offline, our tests show that, with this change included and with the prometheus endpoint enabled, the gcsfuse resource usage does not increase significantly in FIO tests.